composition-js/src/__tests__/compose.test.ts

```diff
@@ -221,6 +221,43 @@ describe('composition', () => {
     `);
   })
 
+  it('preserves descriptions for executable directives', () => {
+    const subgraph1 = {
+      name: 'Subgraph1',
+      typeDefs: gql`
+        "The foo directive description"
+        directive @foo(url: String) on FIELD
+
+        schema {
+          query: Query
+        }
+
+
+        type Query {
+          t(x: String!): String
+        }
+      `
+    }
+
+    const subgraph2 = {
+      name: 'Subgraph2',
+      typeDefs: gql`
+        "The foo directive description"
+        directive @foo(url: String) on FIELD
+
+        enum E {
+          A
+          B
+        }
+      `
+    }
+
+    const result = composeAsFed2Subgraphs([subgraph1, subgraph2]);
+    assertCompositionSuccess(result);
+
+    expect(result.supergraphSdl).toMatchSnapshot();
```

Comment:
Not necessarily a blocker as I suppose it's subjective, but not a huge fan of external snapshots: it makes it unnecessarily hard to follow what the test does imo.

But also, I'd really prefer avoiding matching the whole supergraph SDL as done here since this test is not about the overall supergraph underlying format. Matching everything means that unrelated changes (a bump of some spec version, a new directive definition added to the supergraph, etc...) will require modifying this snapshot.

Here we can easily match the API schema instead (as done in the previous test).